core/sql/generator/GenPreCode.cpp

```diff
@@ -4368,14 +4368,6 @@ RelExpr * GenericUpdate::preCodeGen(Generator * generator,
 	  identityCol = valId.getNAColumn();
 	}
 
-      if ((getOperatorType() != REL_HBASE_UPDATE) &&
-	  (mergeInsertRecExpr().entries() > 0))
-	{
-	  *CmpCommon::diags() << DgSqlCode(-3241)
-			      << DgString0(" Non-unique ON clause not allowed with INSERT.");
-	  GenExit();
-	}
-
```

Comment:
Yes, some of the 3241 errors remain in the generator. I had to move this particular test up to the binder because with the costing changes I was sometimes failing to get a plan in phase 1 of optimization.